src/scenic/core/regions.py

```diff
@@ -281,883 +306,2816 @@ def __hash__(self):
+        position, heading, hw, hl = self.position, self.heading, self.hw, self.hl
+        assert not any(isLazy(c) for c in (position, heading, hw, hl))
+        corners = _RotatedRectangle.makeCorners(position.x, position.y, heading, hw, hl)
+        polygon = shapely.geometry.Polygon(corners)
+        return MultiPolygon([polygon])
+
+
+    def sampleGiven(self, value):
+        return RectangularRegion(value[self.position], value[self.heading],
+            value[self.width], value[self.length],
+            name=self.name)
+
+    def evaluateInner(self, context):
+        position = valueInContext(self.position, context)
+        heading = valueInContext(self.heading, context)
+        width = valueInContext(self.width, context)
+        length = valueInContext(self.length, context)
+        return RectangularRegion(position, heading, width, length, name=self.name)
+
+    def uniformPointInner(self):
+        hw, hl = self.hw, self.hl
+        rx = random.uniform(-hw, hw)
+        ry = random.uniform(-hl, hl)
+        pt = self.position.offsetRotated(self.heading, Vector(rx, ry, self.position.z))
+        return self.orient(pt)
+
+    @property
+    def AABB(self):
+        x, y, z = zip(*self.corners)
+        minx, maxx = findMinMax(x)
+        miny, maxy = findMinMax(y)
+        return ((minx, miny), (maxx, maxy), (self.z, self.z))
+
+    def __repr__(self):
+        return f'RectangularRegion({self.position!r},{self.heading!r},{self.width!r},{self.length!r})'
+
+class PolylineRegion(Region):
+    """Region given by one or more polylines (chain of line segments).
+
+    The region may be specified by giving either a sequence of points or ``shapely``
+    polylines (a ``LineString`` or ``MultiLineString``).
+
+    Args:
+        points: sequence of points making up the polyline (or `None` if using the
+            **polyline** argument instead).
+        polyline: ``shapely`` polyline or collection of polylines (or `None` if using
+            the **points** argument instead).
+        orientation (optional): :term:`preferred orientation` to use, or `True` to use an
+            orientation aligned with the direction of the polyline (the default).
+        name (str; optional): name for debugging.
+    """
+    def __init__(self, points=None, polyline=None, orientation=True, name=None):
+        if orientation is True:
+            orientation = VectorField('Polyline', self.defaultOrientation)
+            self.usingDefaultOrientation = True
+        else:
+            self.usingDefaultOrientation = False
+        super().__init__(name, orientation=orientation)
+
+        if points is not None:
+            points = tuple(toVector(pt).coordinates for pt in points)
+
+            if any(point[2] != 0 for point in points):
+                warnings.warn('"points" passed to PolylineRegion have nonzero Z'
+                              ' components. These will be replaced with 0.')
+
+            points = tuple((x,y,0) for x,y,_ in points)
```

Comment:
It seems inconsistent to allow a `z` parameter for `PolygonalRegion` but not `PolylineRegion`.